lib/ansible/modules/packaging/os/zypper.py

```diff
@@ -107,6 +108,18 @@
         required: false
         default: "no"
         choices: [ "yes", "no" ]
+    retries:
+        version_added: "2.3"
+        description:
+            - Number of retries if another instance of zypper is running. Defaults to no retries.
+        required: false
+        default: 0
+    retry_interval:
+        version_added: "2.3"
+        description:
+            - Time in seconds to wait between retries.
+        required: false
+        default: 10
```

Comment:
The `until` feature should be able to do the same.

Earlier I thought there was more logic needed, but it turns out this just needs to look for `rc==7` and rerun, which is also possible outside of the module. I still think it's ok to have this in the module, since zypper has a bit unusual return codes and instead of having the user find out which code should lead to which workaround the module can just handle this. We already do reruns on `rc==103`. In zypper the database is locked for reads as well, such that a return code of 7 is really common and there should be an obvious way to work around that. 
@resmo even asked to make retries default, which would not be possible with the `until` feature.